src/Razor/src/Microsoft.VisualStudio.Editor.Razor/DefaultVisualStudioMacDocumentInfoFactory.cs

```diff
@@ -20,7 +20,7 @@ internal class DefaultVisualStudioMacDocumentInfoFactory : VisualStudioMacDocume
     {
         public override DocumentInfo CreateEmpty(string razorFilePath, ProjectId projectId)
         {
-            var filename = Path.ChangeExtension(razorFilePath, ".g.cs");
+            var filename = razorFilePath + ".g.cs";
```

Comment:
This is done absolutely everywhere which is awful. We should spend time to make this a factory to normalize this construction. I've filed [this](https://github.com/dotnet/razor-tooling/issues/6630) to track that work.